packages/wallet/src/KeyManagement/types.ts

```diff
@@ -1,19 +1,39 @@
-import { AddressType } from '..';
 import { CSL, Cardano } from '@cardano-sdk/core';
 import { TxInternals } from '../Transaction';
 
+/** Internal = change address & External = receipt address */
+export enum AddressType {
+  /**
+   * Change address
+   */
+  Internal = 1,
+  /**
+   * Receipt address
+   */
+  External = 0
+}
+
+export interface GroupedAddress {
+  networkId: Cardano.NetworkId;
+  type: AddressType;
+  accountIndex: number;
+  addressIndex: number;
+  address: Cardano.Address;
+  rewardAccount: Cardano.RewardAccount;
+}
+
 export interface KeyManager {
-  deriveAddress: (addressIndex: number, index: 0 | 1) => string;
+  // TODO: do not expose CSL objects publicly
+  // Reconsider which keys should be exposed, if any.
```

Comment:
The public keys should be accessible, and `publicAccountKey` is good as-is for now, but `publicStakeKey` should be probably be returned based on the address derivation path. This is how `cardano-wallet` handles it: https://input-output-hk.github.io/cardano-wallet/api/edge/#tag/Keys

We can map this through from the `Wallet` interface to keep the KeyManager an internal concern.